services/worker/src/worker/asset.py

```diff
@@ -58,6 +60,34 @@ class AudioSource(TypedDict):
     type: str
 
 
+def create_audio_files_from_bytes(
+    dataset: str,
+    config: str,
+    split: str,
+    row_idx: int,
+    column: str,
+    array: bytearray,
+    assets_base_url: str,
+    filename_base: str,
+    assets_directory: StrPath,
+) -> List[AudioSource]:
+    wav_filename = f"{filename_base}.wav"
+    mp3_filename = f"{filename_base}.mp3"
+    dir_path, url_dir_path = create_asset_dir(
+        dataset=dataset, config=config, split=split, row_idx=row_idx, column=column, assets_directory=assets_directory
+    )
+    wav_file_path = dir_path / wav_filename
+    mp3_file_path = dir_path / mp3_filename
+    array, sampling_rate = soundfile.read(io.BytesIO(array))
```

Comment:
this is the only difference between this new method and create_audio_files. I think we should have only one method supporting various types, or at least this method should just extract the array and sampling rate, and call the other method.

Reply:
Agree, I fixed it on https://github.com/huggingface/datasets-server/pull/988/commits/b2f428b520bf4799497543ca3692920493529cd1  thanks!.